### PR TITLE
Add CommandBar browser destination

### PR DESCRIPTION
**Pull request by tkainrad**
## CommandBar Browser Destination Overview

We are thrilled to submit our new device mode browser destination. [CommandBar](https://www.commandbar.com/) provides a set of widgets to eliminate UX friction.

With this destination, we want to enable users to 

- initialize CommandBar
- add user `traits` to CommandBar on `identify` -> `identifyUser` action
     - this will allow users to control CommandBar behaviour based on these tratis 
- pass `track` calls CommandBar -> `trackEvent` action
    - this will let users trigger specific CommandBar behaviour like showing nudges

### Settings

```
{
    "orgId": "84f6747a",
    "deploy": true,
    "subscriptions": [{
            "name": "Identify User",
            "subscribe": "type = \"identify\"",
            "partnerAction": "identifyUser",
            "enabled": true,
            "mapping": {
                "userId": {
                    "@path": "$.userId"
                },
                "traits": {
                    "@path": "$.traits"
                },
                "hmac": {
                    "@path": "$.context.CommandBar.hmac"
                },
                "formFactor": {
                    "@path": "$.context.CommandBar.formFactor"
                }
            }
        },
        {
            "partnerAction": "trackEvent",
            "name": "Track Event",
            "enabled": true,
            "subscribe": "type = \"track\"",
            "mapping": {
                "event_name": {
                    "@path": "$.event"
                },
                "event_metadata": {
                    "@path": "$.properties"
                }
            }
        }
    ]
}
```

### Events

**Identify**

```json
{
    "userId": "test-base-user",
    "traits": {
        "foo": "bar"
    },
    "options": {
        "CommandBar": {
            "hmac": "cc9185c5d19062c92798472edec639ee20826aa048b8aa77fe868f556c849d93"
        }
    }
}
```

**Track**

```json
{
    "name": "test-event",
    "properties": {
        "foo": "bar"
    }
}
```

image

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

---

Let me know if you have any questions.

Thanks for creating this framework with its powerful testing environment; it has been great to work with!

### CODE REVIEW

File: packages/browser-destinations/src/destinations/commandbar/index.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import trackEvent from './trackEvent'
+
+declare global {
+  interface Window {
+    CommandBar: CommandBarClientSDK
+  }
+}
+
+export const destination: BrowserDestinationDefinition<Settings, CommandBarClientSDK> = {
+  name: 'Command Bar',
```
Comment (tcgilbert) [inline packages/browser-destinations/src/destinations/commandbar/index.ts:19]: Noting that for your source integration the "name" used is "CommandBar" without the space. Would you like to update the name here to keep this congruent?

Comment (lchoward) [reply packages/browser-destinations/src/destinations/commandbar/index.ts:19]: Ah great catch, yes!

Comment (tkainrad) [reply packages/browser-destinations/src/destinations/commandbar/index.ts]: Oh wow, not sure how that came to pass. Thanks for fixing it in 7d78599137075e7cdf2126938b57358a11eab7f7 @lchoward

File: packages/browser-destinations/src/destinations/commandbar/index.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+declare global {
+  interface Window {
+    CommandBar: CommandBarClientSDK
+  }
+}
+
+export const destination: BrowserDestinationDefinition<Settings, CommandBarClientSDK> = {
+  name: 'CommandBar',
+  slug: 'commandbar',
```
Comment (pooyaj) [inline packages/browser-destinations/src/destinations/commandbar/index.ts:20]:
```suggestion
  slug: 'actions-commandbar',
```

File: packages/browser-destinations/src/destinations/commandbar/identifyUser/index.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import type { BrowserActionDefinition } from '../../../lib/browser-destinations'
+import type { Settings } from '../generated-types'
+import { CommandBarClientSDK, FormFactorConfig, InstanceAttributes } from '../types'
+import type { Payload } from './generated-types'
+
+const action: BrowserActionDefinition<Settings, CommandBarClientSDK, Payload> = {
+  title: 'Identify User',
+  description: 'Set attributes for the user in CommandBar. If "Deploy via Segment" is enabled, then also boot CommandBar for the user, which makes CommandBar available to the user.',
+  platform: 'web',
```
Comment (pooyaj) [inline packages/browser-destinations/src/destinations/commandbar/identifyUser/index.ts:9]: Can we add a: 
```
defaultSubscription: 'type = "identify"',
```
To the action definition?

File: packages/browser-destinations/src/destinations/commandbar/trackEvent/index.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import type { BrowserActionDefinition } from '../../../lib/browser-destinations'
+import type { Settings } from '../generated-types'
+import { CommandBarClientSDK } from '../types'
+import type { Payload } from './generated-types'
+
+const action: BrowserActionDefinition<Settings, CommandBarClientSDK, Payload> = {
+  title: 'Track Event',
+  description: "Submit an event's properties as CommandBar metaData.",
+  platform: 'web',
+
```
Comment (pooyaj) [inline packages/browser-destinations/src/destinations/commandbar/trackEvent/index.ts:10]: Similiarly, let's add 
```
defaultSubscription: 'type = "track"',
```

**Review by pooyaj — COMMENTED** [review]
Looks great 🎉 added few minor comments

File: packages/browser-destinations/src/destinations/commandbar/index.ts (hunk 2 above)
Comment (tkainrad) [reply packages/browser-destinations/src/destinations/commandbar/index.ts]: Thanks for the suggestion, applied this change in c1e9a89e.

File: packages/browser-destinations/src/destinations/commandbar/identifyUser/index.ts (hunk 3 above)
Comment (tkainrad) [reply packages/browser-destinations/src/destinations/commandbar/identifyUser/index.ts:9]: Ah, I missed that. Added in c1e9a89e

File: packages/browser-destinations/src/destinations/commandbar/trackEvent/index.ts (hunk 4 above)
Comment (tkainrad) [reply packages/browser-destinations/src/destinations/commandbar/trackEvent/index.ts:11]: Added in c1e9a89e

**Comment by pooyaj** [comment]
@tkainrad running the lint/test on the CI, looks like there are two lint errors: 
```
Run yarn lint
yarn run v1.22.19
$ eslint '**/*.ts' --cache

/home/runner/work/action-destinations/action-destinations/packages/browser-destinations/src/destinations/commandbar/identifyUser/index.ts
Error:   54:9  error  Redundant double negation                                                                        no-extra-boolean-cast
Error:   55:[7](https://github.com/segmentio/action-destinations/actions/runs/3602080625/jobs/6073739946#step:9:8)  error  Promises must be handled appropriately or explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/runner/work/action-destinations/action-destinations/packages/browser-destinations/src/destinations/commandbar/trackEvent/index.ts
Error:   34:[9](https://github.com/segmentio/action-destinations/actions/runs/3602080625/jobs/6073739946#step:9:10)  error  Redundant double negation  no-extra-boolean-cast
```

**Comment by tkainrad** [comment]
> @tkainrad running the lint/test on the CI, looks like there are two lint errors:
> 
> ```
> Run yarn lint
> yarn run v1.22.19
> $ eslint '**/*.ts' --cache
> 
> /home/runner/work/action-destinations/action-destinations/packages/browser-destinations/src/destinations/commandbar/identifyUser/index.ts
> Error:   54:9  error  Redundant double negation                                                                        no-extra-boolean-cast
> Error:   55:[7](https://github.com/segmentio/action-destinations/actions/runs/3602080625/jobs/6073739946#step:9:8)  error  Promises must be handled appropriately or explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
> 
> /home/runner/work/action-destinations/action-destinations/packages/browser-destinations/src/destinations/commandbar/trackEvent/index.ts
> Error:   34:[9](https://github.com/segmentio/action-destinations/actions/runs/3602080625/jobs/6073739946#step:9:10)  error  Redundant double negation  no-extra-boolean-cast
> ```

Thanks for the quick re-review! Sorry to have missed that. Fixed now with 9c4936c3

**Review by pooyaj — APPROVED** [review]

**Pull request merged**